server/api/branch/tests.py

```diff
@@ -33,3 +34,132 @@ def test_GetAllExceptis_deleted(self):
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['name'], 'Test1')
+
+        # "does not exist" case
+        response = self.client.get('/api/branch/4/')
+        self.assertEqual(response.status_code, 404)
+
+    def test_post_branch(self):
+        response = self.client.post('/api/branch/', {
+            'name': 'Test4',
+            'description': 'Desc4'
+        })
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.data['name'], 'Test4')
+
+        # "missing field" case
+        response = self.client.post('/api/branch/', {
+            'name': 'Test4'
+        })
+        self.assertEqual(response.status_code, 400)
+
+        # "empty field" case
+        response = self.client.post('/api/branch/', {
+            'name': '',
+            'description': 'Desc4'
+        })
+        self.assertEqual(response.status_code, 400)
+
+        # "unauthorised" case
+        response = self.client.post('/api/branch/4/', {
+            'name': 'Test5',
+            'description': 'Desc5'
+        })
+        self.assertEqual(response.status_code, 405)
+
+    def test_put_branch(self):
+        response = self.client.put(f'/api/branch/{self.branch.id}/', {
+            'name': 'Test5',
+            'description': 'Desc5'
+        })
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['name'], 'Test5')
+
+        # "missing field" case
+        response = self.client.put(f'/api/branch/{self.branch.id}/', {
+            'name': 'Test5'
+        })
+        self.assertEqual(response.status_code, 400)
+
+        # "empty field" case
+        response = self.client.put(f'/api/branch/{self.branch.id}/', {
+            'name': '',
+            'description': 'Desc5'
+        })
+        self.assertEqual(response.status_code, 400)
+
+        # "invalid field" case
+        response = self.client.put(f'/api/branch/{self.branch.id}/', {
+            'name': 'Test6',
+            'invalid': 'invalid'
+        })
+        self.assertEqual(response.status_code, 400)
+
+        # "does not exist" case
+        response = self.client.put('/api/branch/4/', {
+            'name': 'Test5',
+            'description': 'Desc5'
+        })
+        self.assertEqual(response.status_code, 404)
+
+    def test_patch_branch(self):
+        response = self.client.patch(f'/api/branch/{self.branch.id}/', {
+            'name': 'Test6'
+        })
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['name'], 'Test6')
+
+        # "empty field" case
+        response = self.client.patch(f'/api/branch/{self.branch.id}/', {
+            'name': ''
+        })
+        self.assertEqual(response.status_code, 400)
+
+        # "invalid field" case
+        response = self.client.patch(f'/api/branch/{self.branch.id}/', {
+            'invalid': 'invalid'
+        })
+        self.assertEqual(response.status_code, 400)
```

Comment:
It seems that the default behaviour for the UpdateModelMixin is to ignore extra fields when PATCHing. Instead, you could test an "invalid field" by attempting to update a field with the wrong type. Otherwise it's fine to 200

Reply:
Ah ok, DRF decided that's not a biggie, ig. Thanks for looking that up!